### DISCUSSION TITLE
Trouble with testing Python embedding use cases on WCOSS2 Acorn

### DISCUSSION THREAD

**Discussion by jprestop** [dtcenter/METplus#2223] [Use Cases] [answered]
I recently installed METplus-5.1.0-rc1 on WCOSS2 Acorn and tried to test the installation with two Python embedding use cases:

1. run_metplus.py ${METPLUS_PATH}/parm/use_cases/met_tool_wrapper/GridStat/GridStat_python_embedding.conf /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/jpresto.acorn.conf
2. run_metplus.py ${METPLUS_PATH}/parm/use_cases/met_tool_wrapper/EnsembleStat/EnsembleStat_python_embedding.conf /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/jpresto.acorn.conf

Both tests result in the same error:
```
julie.prestopnik@alogin01:/apps/sw_review/emc/METplus> run_metplus.py ${METPLUS_PATH}/parm/use_cases/met_tool_wrapper/GridStat/GridStat_python_embedding.conf /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/jpresto.acorn.conf
Traceback (most recent call last):
  File "/apps/sw_review/emc/METplus/METplus-5.1.0-rc1/ush/run_metplus.py", line 123, in <module>
    produtil.setup.setup(send_dbn=False, jobname='run-METplus')
  File "/apps/sw_review/emc/METplus/METplus-5.1.0-rc1/produtil/setup.py", line 79, in setup
    produtil.cluster.where() # guess cluster
  File "/apps/sw_review/emc/METplus/METplus-5.1.0-rc1/produtil/cluster.py", line 100, in where
    here=WCOSS2()
NameError: name 'WCOSS2' is not defined

ERROR: run_metplus  failed: name 'WCOSS2' is not defined
julie.prestopnik@alogin01:/apps/sw_review/emc/METplus> run_metplus.py ${METPLUS_PATH}/parm/use_cases/met_tool_wrapper/EnsembleStat/EnsembleStat_python_embedding.conf /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/jpresto.acorn.conf
Traceback (most recent call last):
  File "/apps/sw_review/emc/METplus/METplus-5.1.0-rc1/ush/run_metplus.py", line 123, in <module>
    produtil.setup.setup(send_dbn=False, jobname='run-METplus')
  File "/apps/sw_review/emc/METplus/METplus-5.1.0-rc1/produtil/setup.py", line 79, in setup
    produtil.cluster.where() # guess cluster
  File "/apps/sw_review/emc/METplus/METplus-5.1.0-rc1/produtil/cluster.py", line 100, in where
    here=WCOSS2()
NameError: name 'WCOSS2' is not defined

ERROR: run_metplus  failed: name 'WCOSS2' is not defined
```

I have not encountered this error previously. I ran these same tests successfully for the last official installation (METplus-5.0.1).

Here are the contents of my /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/jpresto.acorn.conf file:
```
[dir]
INPUT_BASE = /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/METplus-5.1_sample_data
OUTPUT_BASE = /lfs/h1/emc/vpppg/noscrub/Julie.Prestopnik/METplus/output
#MET_PYTHON_EXE = `which python3`
```
The two other configurations files have not been modified in any way and are the default files.

Can you please help me resolve this problem?

**Comment by jprestop** [comment]
@georgemccabe It looks like the produtil/cluster.py file was touched in a SonarQube related change:
https://github.com/dtcenter/METplus/commit/042b3948528e97b60d1361e6d582fc3bcc94396c#diff-3e673e9c0ce826ed8daa345175ebe347489cbf5b170184fffd5e4e485ffeef61

from a commit here: Feature https://github.com/dtcenter/METplus/issues/1610 first round of SonarQube fixes (https://github.com/dtcenter/METplus/pull/2200)

In case that is helpful...  I'm hoping you'll be familiar with the change and will know how to solve this problem.

**Reply by jprestop** [reply]
@georgemccabe and one more piece of information - I just installed and tested on Hera and everything runs fine on Hera.

**Comment by georgemccabe** [comment, marked as answer]
It looks like there was a 1-line fix needed to resolve this issue. Before I started on resolving the SonarQube complaints, I pulled new changes from NOAA's version of produtil to see if any of the complaints were already handled on their end. I'm not sure how this class rename did not make it in with the rest of the changes from that file. Perhaps there was a merge conflict that was improperly handled by Git? Either way, there is a fix in the develop branch. @jprestop will test that the fix works and we can create METplus v5.1.0-rc2 if all goes smoothly.

**Reply by jprestop** [reply]
Thanks so much, @georgemccabe.  That did the trick!  I grabbed the code from the repository and checked out the develop branch.  I ran the test cases and they were both successful.

**Reply by jprestop** [reply]
Just following up here to link up the issue: [Bugfix: Fix produtil bug for WCOSS #2227](https://github.com/dtcenter/METplus/issues/2227)
Thanks for creating the issue and resolving the bug @georgemccabe.
